src/ElasticSearch/Index.php

```diff
@@ -86,16 +86,32 @@ public function config(): array
     public static function fromSource(ImportSource $source): Index
```

Comment:
I think it will be better to extract index building to some factory and inject it through di container. There can be other cases on how to build settings and mappings and I don't think it's a good place for them.